CMakeLists.txt

```diff
@@ -22,20 +22,15 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 
 if(APPLE)
-    # Check if the target architecture is arm64
-    include(CheckCXXSourceRuns)
-    set(CMAKE_REQUIRED_FLAGS "-arch arm64")
-    set(CHECK_ARM64_SOURCE_CODE "int main() { return 0; }")
-    check_cxx_source_runs("${CHECK_ARM64_SOURCE_CODE}" IS_ARM64)
-    if(IS_ARM64)
+    option (MACOS_BUILD_ARM64 "Build for arm64 (ON) or x86 (OFF)" ON)
+    if(MACOS_BUILD_ARM64)
         message("Building on macOS M1 architecture (arm64)")
-        # Set a flag or perform actions specific to M1 architecture
-        set(M1_ARCHITECTURE TRUE CACHE BOOL "Building on macOS M1 architecture")
         set(CMAKE_OSX_DEPLOYMENT_TARGET "12.0" CACHE STRING "Minimum OS X deployment version" FORCE)
-
+        set(CMAKE_OSX_ARCHITECTURES "arm64")
     else()
         message("Building on macOS x86 architecture")
         set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum OS X deployment version" FORCE)
+        set(CMAKE_OSX_ARCHITECTURES "x86_64" CACHE STRING "macOS build architecture" FORCE)
```

Comment:
We should target the host architecture by default.

Reply:
Ok, native build platforms are now targeted by default.  We can now target specific architectures by specifying CMAKE_OSX_ARCHITECTURES.

Reply:
Great, thank you!